app/models/ability.rb

```diff
@@ -0,0 +1,34 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
```

Comment:
This line here deserves some attention. This is a common idiom used in ruby.  `var ||= Class.new`